hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAirdropStore.java

```diff
@@ -0,0 +1,135 @@
+     */
+    private final WritableKVState<PendingAirdropId, AccountPendingAirdrop> airdropState;
+
+    /**
+     * Create a new {@link WritableAirdropStore} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableAirdropStore(
+            @NonNull final WritableStates states,
+            @NonNull final Configuration configuration,
+            @NonNull final StoreMetricsService storeMetricsService) {
+        super(states);
+        airdropState = states.get(AIRDROPS_KEY);
+
+        final long maxCapacity = configuration.getConfigData(TokensConfig.class).maxAllowedAirdrops();
+        final var storeMetrics = storeMetricsService.get(StoreMetricsService.StoreType.AIRDROP, maxCapacity);
+        airdropState.setMetrics(storeMetrics);
+    }
+
+    /**
+     * Persists a new {@link PendingAirdropId} with given {@link AccountPendingAirdrop} into the state,
+     * as well as exporting its ID to the transaction receipt. If there is existing
+     * airdrop with the same id we add the value to the existing drop.
+     *
+     * @param airdropId    - the airdropId to be persisted.
+     * @param accountAirdrop - the account airdrop mapping for the given airdropId to be persisted.
+     */
+    public void put(@NonNull final PendingAirdropId airdropId, @NonNull final AccountPendingAirdrop accountAirdrop) {
+        requireNonNull(airdropId);
+        requireNonNull(accountAirdrop);
+
+        if (!airdropState.contains(airdropId)) {
+            airdropState.put(airdropId, accountAirdrop);
+            return;
+        }
+
+        if (airdropId.hasFungibleTokenType()) {
+            final var existingAccountAirdrop = requireNonNull(airdropState.get(airdropId));
+            final var existingAirdropValue =
+                    requireNonNull(airdropState.get(airdropId)).pendingAirdropValue();
+            requireNonNull(existingAirdropValue);
+            requireNonNull(accountAirdrop.pendingAirdropValue());
+            final var newValue = accountAirdrop.pendingAirdropValue().amount() + existingAirdropValue.amount();
+            final var newAccountAirdrop = existingAccountAirdrop
+                    .copyBuilder()
+                    .pendingAirdropValue(
+                            PendingAirdropValue.newBuilder().amount(newValue).build())
+                    .build();
+            airdropState.put(airdropId, newAccountAirdrop);
+        }
+    }
+
+    /**
+     * Updates given {@link PendingAirdropId} with new {@link AccountPendingAirdrop} value the state,
+     * as well as exporting its ID to the transaction receipt. If there is no existing
+     * airdrop with the same id do nothing.
+     *
+     * @param airdropId    - the airdropId to be updated.
+     * @param accountAirdrop - the account airdrop mapping for the given airdropId to be updated.
+     */
+    public void patch(@NonNull final PendingAirdropId airdropId, @NonNull final AccountPendingAirdrop accountAirdrop) {
```

Comment:
Do we need this ?

Reply:
Here is Ivan's answer to this question on the old PR: https://github.com/hashgraph/hedera-services/pull/13830#discussion_r1677644966

P.S.. we will go through the old PR comments and address them.

Reply:
With the current put method, we don't have a way to replace existing AccountPendingAirdrop in state with a different one without adding the values, we either have to have this patch method or move the logic where we combine the values of fungible airdrops to the handler instead of the store.

Reply:
Oh Okay! If its not needed can be removed later also.